app/controllers/notifications_controller.rb

```diff
@@ -122,6 +123,16 @@ def expand_comments
     end
   end
 
+  def comment
+    subject = current_user.notifications.find(params[:id]).subject
+    subject.comment(current_user, params[:comment][:body]) if subject.commentable?
```

Comment:
We should check if the user `can_comment?` here before attempting it, and then redirect with an error if they cannot.